response_operations_ui/views/collection_exercise.py

```diff
@@ -13,10 +13,11 @@
 
 
 @app.route('/surveys/<short_name>/<period>', methods=['GET'])
-def view_collection_exercise(short_name, period):
+def view_collection_exercise(short_name, period, error={}, ci_loaded=False, sample_loaded=False):
```

Comment:
It's usually unsafe to have a mutable object (`error={}`) as a default parameter to a function as the same object can be used multiple times for multiple calls (created once at function definition time).

Maybe change to `error=None` and then in the return `error=error or {}` or `error={} if error is None else error` or similar.